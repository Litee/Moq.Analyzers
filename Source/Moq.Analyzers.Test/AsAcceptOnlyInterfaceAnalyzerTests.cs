namespace Moq.Analyzers.Test
{
    using System.IO;
    using ApprovalTests;
    using ApprovalTests.Reporters;
    using Microsoft.CodeAnalysis.Diagnostics;
    using TestHelper;
    using Xunit;

    [UseReporter(typeof(DiffReporter))]
    public class AsAcceptOnlyInterfaceAnalyzerTests : DiagnosticVerifier
    {
        [Fact]
        public void ShouldPassIfGoodParameters()
        {
            Approvals.Verify(VerifyCSharpDiagnostic(File.ReadAllText("Data/AsAcceptOnlyInterface.cs")));
        }

        protected override DiagnosticAnalyzer GetCSharpDiagnosticAnalyzer()
        {
            return new AsShouldBeUsedOnlyForInterfaceAnalyzer();
        }
    }
}