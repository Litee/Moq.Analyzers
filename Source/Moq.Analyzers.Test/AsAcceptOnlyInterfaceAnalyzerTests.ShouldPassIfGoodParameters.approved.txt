Diagnostic 1
	Id: Moq1300
	Location: SourceFile(Test0.cs[1559..1574))
	Highlight: BaseSampleClass
	Lines: mock.As<BaseSampleClass>();
	Severity: Error
	Message: Mock.As() should take interfaces only

Diagnostic 2
	Id: Moq1300
	Location: SourceFile(Test0.cs[1730..1740))
	Highlight: OtherClass
	Lines: mock.As<OtherClass>();
	Severity: Error
	Message: Mock.As() should take interfaces only

